lib/experimental/interactivity-api/navigation-block-interactivity.php

```diff
@@ -74,21 +74,6 @@ function gutenberg_block_core_navigation_add_directives_to_markup( $block_conten
 		// If the open modal button not found, we handle submenus immediately.
 		$w = new WP_HTML_Tag_Processor( $w->get_updated_html() );
```

Comment:
I think we could also apply the following change here:

```diff
diff --git a/lib/experimental/interactivity-api/navigation-block-interactivity.php b/lib/experimental/interactivity-api/navigation-block-interactivity.php
index fd024d0c92..e782d6b1ae 100644
--- a/lib/experimental/interactivity-api/navigation-block-interactivity.php
+++ b/lib/experimental/interactivity-api/navigation-block-interactivity.php
@@ -72,7 +72,10 @@ function gutenberg_block_core_navigation_add_directives_to_markup( $block_conten
 		$w->remove_attribute( 'data-micromodal-trigger' );
 	} else {
 		// If the open modal button not found, we handle submenus immediately.
-		$w = new WP_HTML_Tag_Processor( $w->get_updated_html() );
+		$w = new WP_HTML_Tag_Processor( $block_content );
+		if ( $w->next_tag( 'nav' ) ) {
+			$w->set_attribute( 'data-wp-island', '' );
+		}
 
 		gutenberg_block_core_navigation_add_directives_to_submenu( $w );
```

The reason for that is the context set assumes we use an overlay which is not true:

```php
$w->set_attribute( 'data-wp-context', '{ "core": { "navigation": { "isMenuOpen": false, "overlay": true, "roleAttribute": "" } } }' );
```

I'm not sure why it didn't work for me without the changes applied to the `UL` tag before with the following block settings:

<img width="279" alt="Screenshot 2023-05-18 at 10 54 42" src="https://github.com/WordPress/gutenberg/assets/699132/67d021f1-deeb-4348-9fa0-9871f899e174">

However, I can confirm that everything works as expected with this branch 🎉

Reply:
> I think we could also apply the following change here:

Not sure if we need that. At that point, in the parsing we are already handling the submenus, and there is no `nav` tag if I am not mistaken. We will always have a parent `nav` even if the overlay is "off", and we are already adding the `data-wp-island` [here](https://github.com/WordPress/gutenberg/blob/fix/polish-experimental-navigation-block/lib/experimental/interactivity-api/navigation-block-interactivity.php#L58-L61). It is true that we are adding the `overlay: true` to the parent `nav`, but it shouldn't affect as we are overwriting it in [the submenus context](https://github.com/WordPress/gutenberg/blob/fix/polish-experimental-navigation-block/lib/experimental/interactivity-api/navigation-block-interactivity.php#L176).

Does this make sense?

> I'm not sure why it didn't work for me without the changes applied to the UL tag before with the following block settings

I believe it was happening because we didn't have [the `else` statement to jump into the submenus](https://github.com/WordPress/gutenberg/blob/fix/polish-experimental-navigation-block/lib/experimental/interactivity-api/navigation-block-interactivity.php#L73-L80). This means it was trying to find [the overlay container](https://github.com/WordPress/gutenberg/blob/fix/polish-experimental-navigation-block/lib/experimental/interactivity-api/navigation-block-interactivity.php#L82-L96) and the next tags (not found) and when it tried to handle the submenu the parser had already passed that part I guess.

Reply:
> It is true that we are adding the overlay: true to the parent nav, but it shouldn't affect as we are overwriting it in [the submenus context](https://github.com/WordPress/gutenberg/blob/fix/polish-experimental-navigation-block/lib/experimental/interactivity-api/navigation-block-interactivity.php#L176).

Got it. No changes needed then 👍🏻